azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -53,10 +54,12 @@
   private final ProjectLoader projectLoader;
   private final Props props;
   private final boolean creatorDefaultPermissions;
+  // Both projectsById and projectsByName cache should use the concurrent data structure to
```

Comment:
Thanks for adding the comment.

This is good. 

What do you think of an alternative?

"Both projectsById and projectsByName cache need to be thread safe since they are accessed from multiple threads concurrently without external synchronization for performance."

Reply:
Updated, thanks!